### ISSUE TITLE
Prevent building with busted xcode tools

### ISSUE THREAD

**Issue by jdm** [servo/servo#22978]
We know that a particular version of xcode that is available with MacOS Mojave contains a bug that prevents linking Servo. We should detect that and tell users attempting to build Servo with that XCode version that they need to update.

It might be possible to use `xcode-select --version` to detect this. On a system with a busted version, the output is `2354`. On my system which is pre-Mojave is it `2347`.

**Comment by jdm** [comment]
@paulrouget What does `xcode-select --version` show on a system with the xcode beta installed?

**Comment by pradyunsg** [comment]
The check has to be added in this file, or is there a better place?

https://github.com/servo/servo/blob/master/python/servo/build_commands.py#L200

**Comment by jdm** [comment]
Yes, that seems fine to me. We have similar checks for known-broken configurations at https://github.com/servo/servo/blob/55347aa39f272b5b3fc7ede6403f70aa49d31ce1/python/servo/build_commands.py#L215-L218.

**Comment by CYBAI** [comment]
I remember I'm using Xcode beta on macOS Mojave (10.14.3) but, unfortunately, I see `2354` from `xcode-select --version`. Not sure if I checked in the right way? 🤔

```bash
→ xcode-select -p
/Applications/Xcode-beta.app/Contents/Developer

→ xcode-select --version
xcode-select version 2354.
```

**Comment by jdm** [comment]
Humbug. What about `dsymutil --version`?

**Comment by CYBAI** [comment]
```
→ dsymutil --version
Apple LLVM version 10.0.1 (clang-1001.0.37.9)
  Optimized build.
  Default target: x86_64-apple-darwin18.2.0
  Host CPU: skylake
```

**Comment by jdm** [comment]
```
Apple LLVM version 9.0.0 (clang-900.0.39.2)
  Optimized build.
  Default target: x86_64-apple-darwin16.7.0
  Host CPU: haswell
```
How about yours @pradyunsg?

**Comment by pradyunsg** [comment]
```
Apple LLVM version 10.0.0 (clang-1000.11.45.5)
  Optimized build.
  Default target: x86_64-apple-darwin18.2.0
  Host CPU: haswell
```

**Comment by pradyunsg** [comment]
Should I try adding homebrew's LLVM to the path? (it does seem insanely old)


```
$ /usr/local/Cellar/llvm/7.0.1/bin/dsymutil --version
LLVM (http://llvm.org/):
  LLVM version 7.0.1
  Optimized build.
  Default target: x86_64-apple-darwin18.2.0
  Host CPU: haswell
```

**Comment by jdm** [comment]
No idea if that will work or not. The supported way is to install the xcode beta.

**Comment by pradyunsg** [comment]
I'll try after my paper today.

**Comment by paulrouget** [comment]
`xcode-select --version` shows `2354` with the beta. 

`dsymutil --version` shows:

```
Apple LLVM version 10.0.1 (clang-1001.0.37.7)
  Optimized build.
  Default target: x86_64-apple-darwin18.2.0
  Host CPU: skylake
```

**Comment by pradyunsg** [comment]
> The supported way is to install the xcode beta.

Alrighty then, starting an ~7 hour download then. :)

**Comment by jschwe** [comment]
This issue should be solved (by virtue of time passing, and nobody using that busted version anymore)
